pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -107,6 +110,20 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// Add the finalizer if it is not present
+	if err := r.setFinalizers(userAcc); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if the UserAccount has been deleted
+	if !userAcc.ObjectMeta.DeletionTimestamp.IsZero() {
+		rec, err := r.finalizer(reqLogger, userAcc)
```

Comment:
should we also check if `err != nil` here ??

Reply:
Refactored the code - hopefully this is alright: 51e17d72b87aa88373f9bdb8dd5dd696b2b29847